### PR TITLE
Neural net

### PR DESCRIPTION
**Pull request by ayushbihani**

### CODE REVIEW

File: blocks/tensorflow.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    ],
+    "previousStatement": null,
+    "nextStatement": null,
+    "inputsInline": false,
+    "colour": 230,
+    "tooltip": "",
+    "helpUrl": ""
+  },
+  {
+    "type": "tf_model_compile",
```
Comment (arjun-rao) [inline blocks/tensorflow.js]: Combine this block with `tf_model` as it's confusing to have a compile block option.

File: blocks/tensorflow.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      }
+    ],
+    "previousStatement": null,
+    "nextStatement": null,
+    "colour": 230,
+    "tooltip": "",
+    "helpUrl": ""
+  },
+  {
+    "type": "tf_model",
```
Comment (arjun-rao) [inline blocks/tensorflow.js]: change this to `tf_neural_network`

File: blocks/tensorflow.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    "message0": "Create Tensorflow Model %1 X (Features) %2 Add layers %3",
```
Comment (arjun-rao) [inline blocks/tensorflow.js]: Instead use `Create Neural Network` or `Define Neural Network`

File: blocks/tensorflow.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        "name": "model"
+      }
+    ],
+    "output": "tfmodel",
+    "colour": 230,
+    "tooltip": "",
+    "helpUrl": ""
+  },
+  {
+    "type": "tf_train_model",
```
Comment (arjun-rao) [inline blocks/tensorflow.js]: Take input and labels here instead of taking input features during model definition.

File: blocks/tfjs.js [hunk 5]
```diff
@@ -212,3 +212,13 @@ Blockly.defineBlocksWithJsonArray([  // BEGIN JSON EXTRACT
   }
 
 ]);
+
+Blockly.Blocks['test'] = {
+  init: function() {
+    this.appendStatementInput("NAME")
+        .setCheck(null);
+    this.setColour(230);
+ this.setTooltip("");
```
Comment (arjun-rao) [inline blocks/tfjs.js]: Indentation mismatch - run `npm run test:lint` before submitting PR

File: blocks/tfjs.js [hunk 6]
```diff
@@ -212,3 +212,13 @@ Blockly.defineBlocksWithJsonArray([  // BEGIN JSON EXTRACT
...
+ this.setHelpUrl("");
+  }
+};
```
Comment (arjun-rao) [inline blocks/tfjs.js]: Add blank line at the end of file

File: blocks/tensorflow.js [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Plot blocks for Milo (github.com/4and4/MiloServer).
```
Comment (arjun-rao) [inline blocks/tensorflow.js]: Should be Neural Network blocks. Change file name to `neuralnet.js`

File: generators/javascript/tensorflow.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Plot blocks for Milo (github.com/4and4/MiloServer).
```
Comment (arjun-rao) [inline generators/javascript/tensorflow.js]: Should be Neural Network Blocks. Change file name to `neuralnet.js`

File: generators/javascript/tensorflow.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+Blockly.JavaScript['tf_model_compile'] = function(block) {
+  var value_model = Blockly.JavaScript.valueToCode(block, 'model', Blockly.JavaScript.ORDER_ATOMIC);
+  var value_rate = Blockly.JavaScript.valueToCode(block, 'rate', Blockly.JavaScript.ORDER_ATOMIC);
+  var dropdown_optimizer = block.getFieldValue('optimizer');
+  var dropdown_loss = block.getFieldValue('loss');
+  var options = 'var options = {\n' +
+    '\n optimizer:getOptimizer("'+dropdown_optimizer+'",'+value_rate+'),'+
+    '\n loss:"'+dropdown_loss+'",'+
+    '\n metrics:["accuracy"]'+
+  '}\n';
```
Comment (arjun-rao) [inline generators/javascript/tensorflow.js]: Change line to include additional `\n` in the beginning: 
```
`\n}\n`;
```

File: generators/javascript/tensorflow.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  var code = value_model+'.compileModel(options)\n';
+  return [options,code].join("\n");
+};
+
+Blockly.JavaScript['tf_model'] = function(block) {
```
Comment (arjun-rao) [inline generators/javascript/tensorflow.js]: change this to `tf_neural_network` and combine tf_model_options with this block - so that there's no compile block. Move the training set input features to the tf_model_train block - and rename that to `tf_neural_network_train`

File: generators/javascript/tensorflow.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+Blockly.JavaScript['tf_train_model'] = function(block) {
+  var value_model = Blockly.JavaScript.valueToCode(block, 'model', Blockly.JavaScript.ORDER_ATOMIC);
+  //var value_x = Blockly.JavaScript.valueToCode(block, 'x', Blockly.JavaScript.ORDER_ATOMIC);
+  var value_y = Blockly.JavaScript.valueToCode(block, 'y', Blockly.JavaScript.ORDER_ATOMIC);
+  var code = value_model +'.train(' +value_y+')\n';
+  return code;
+};
+
+Blockly.JavaScript['tf_model_predict'] = function(block) {
```
Comment (arjun-rao) [inline generators/javascript/tensorflow.js]: Rename to `tf_neural_network_predict`

File: generators/javascript/tfjs.js [hunk 12]
```diff
@@ -71,4 +71,10 @@ Blockly.JavaScript['tf_basic_math'] = function(block) {
     var value_tensor = Blockly.JavaScript.valueToCode(block, 'tensor', Blockly.JavaScript.ORDER_ATOMIC);
     var code = value_tensor+'.'+dropdown_math + '() \n';
     return [code, Blockly.JavaScript.ORDER_NONE];
-};
\ No newline at end of file
+};
+
+Blockly.JavaScript['test'] = function(block) {
```
Comment (arjun-rao) [inline generators/javascript/tfjs.js]: Remove this unused block
